File: frontend/types/notification.ts
// frontend/types/notification.ts

export type NotificationType = "deadline" | "milestone" | "team" | "system";

export interface Notification {
    id: string;
    title: string;
    message: string;
    type: NotificationType;
    is_read: boolean;
    created_at: string;
}

File: backend/app/schemas/notification.py
# backend/app/schemas/notification.py

from pydantic import BaseModel, ConfigDict
from datetime import datetime
from uuid import UUID
from app.models.notification import NotificationType

class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    message: str
    type: NotificationType
    is_read: bool
    created_at: datetime

class NotificationUpdate(BaseModel):
    is_read: bool